src/CG-DNA/atom_vec_oxdna.cpp

```diff
@@ -0,0 +1,96 @@
+   LAMMPS - Large-scale Atomic/Molecular Massively Parallel Simulator
+   https://www.lammps.org/, Sandia National Laboratories
+   Steve Plimpton, sjplimp@sandia.gov
+
+   Copyright (2003) Sandia Corporation.  Under the terms of Contract
+   DE-AC04-94AL85000 with Sandia Corporation, the U.S. Government retains
+   certain rights in this software.  This software is distributed under
+   the GNU General Public License.
+
+   See the README file in the top-level LAMMPS directory.
+------------------------------------------------------------------------- */
+
+#include "atom_vec_oxdna.h"
+#include "atom.h"
+#include "comm.h"
+
+using namespace LAMMPS_NS;
+
+/* ---------------------------------------------------------------------- */
+AtomVecOxdna::AtomVecOxdna(LAMMPS *lmp) : AtomVec(lmp)
+{
+  molecular = Atom::MOLECULAR;
+  bonds_allow = 1;
+  mass_type = PER_TYPE;
+
+  atom->molecule_flag = 1;
+
+  // strings with peratom variables to include in each AtomVec method
+  // strings cannot contain fields in corresponding AtomVec default strings
+  // order of fields in a string does not matter
+  // except: fields_data_atom & fields_data_vel must match data file
+
+  fields_grow = (char *) "id5p";
+  fields_copy = (char *) "id5p";
+  fields_comm = (char *) "";
+  fields_comm_vel = (char *) "";
+  fields_reverse = (char *) "";
+  fields_border = (char *) "id5p";
+  fields_border_vel = (char *) "";
+  fields_exchange = (char *) "id5p";
+  fields_restart = (char *) "id5p";
+  fields_create = (char *) "";
+  fields_data_atom = (char *) "id type x";
+  fields_data_vel = (char *) "id v";
+
+  setup_fields();
+
+}
+
+/* ---------------------------------------------------------------------- */
+AtomVecOxdna::~AtomVecOxdna()
+{
+
+}
+
+/* ----------------------------------------------------------------------
+   set local copies of all grow ptrs used by this class, except defaults
+   needed in replicate when 2 atom classes exist and it calls pack_restart()
+------------------------------------------------------------------------- */
+
+void AtomVecOxdna::grow_pointers()
+{
+  id5p = atom->id5p;
+}
+
+/* ----------------------------------------------------------------------
+   initialize atom quantity 5' partner
+------------------------------------------------------------------------- */
+
+void AtomVecOxdna::data_atom_post(int ilocal)
+{
+  tagint *id5p = atom->id5p;
+  id5p[ilocal] = -1;
+}
+
+/* ----------------------------------------------------------------------
+   process bond information as per data file
+   store 5' partner to inform 3'->5' bond directionality 
+------------------------------------------------------------------------- */
+
+void AtomVecOxdna::data_bonds_post(int m, int num_bond, tagint atom1,
+                                       tagint atom2, tagint id_offset)
+{
+  tagint *id5p = atom->id5p;
+
+  if (id_offset) {
+    atom1 += id_offset;
+    atom2 += id_offset;
+  }
+
+  if ((m = atom->map(atom1)) >= 0) {
```

Comment:
I'm not 100% clear the logic in this if statement is correct, based on what you explained earlier.
(1)you are overwriting the passed input arg "m", which means you are ignoring it.
Even if that's OK, you should use a different variable name.
(2) Can you verity that your code still works and gives the same answers if newton bond is set to off?  I think the code here in that case is simply setting id5p[] twice for the same atom, but it would be good to verify that.  I thought you were planning to skip doing this setting on the 2nd call when newton bond off, but maybe this is simpler.

Reply:
@sjplimp 
Ad (1): I use now a different variable name, which is internally defined. I agree this is better as indeed the argument 'm' was ignored.

Ad(2): The code still works with newton off. During the second call to avec->data_bonds_post() in Atom::data_bonds() with newton off it simply assigns the same values id5p[n]=atom2 with n = atom->map(atom1), so indeed what you say it sets it twice. This is far easier than having a first/second pass criterion, a possibility I didn't see in the first instance.